openstack_openapi/openapi.rst

```diff
@@ -0,0 +1,465 @@
+=======================
+OpenStack OpenAPI specs
+=======================
+
+Building up OpenAPI specs for OpenStack APIs
+--------------------------------------------
+
+:Date: 2024-05-15
+:Author: Artem Goncharov
+
+
+Agenda
+------
+
+- Introduction
+
+- OpenStack API "features"
+
+- Designing OpenAPI for OpenStack
+
+- CodeGenerator
+
+- Rust Tooling |crab|
+
+
+What is OpenAPI
+---------------
+
+The OpenAPI Specification, previously known as the Swagger Specification, is a
+specification for a machine-readable interface definition language for
+describing, producing, consuming and visualizing web services.
+
+. . .
+
+Version 3.1.0
+~~~~~~~~~~~~~
+
+OpenAPI version 3.1.0, released in 2021, includes JSON schema alignment what
+allows it to be applied for every API with JSON schema.
+
+What is OpenAPI good for
+------------------------
+
+-  Machine readable API contract
+-  API documentation consistent with the service code
+-  Generation of API clients
```

Comment:
also generation of API documentation
and generation of API functional tests

I'm also missing some short detail what openapi can contain:
path, response codes, headers, request schema, response schema, mandatory and optional parameters...
Also maybe 1,2 slides with showing real openapi example for single API would be nice

Reply:
adding separate slide with sample.